tests/test_geometry.py

```diff
@@ -1,190 +1,183 @@
+    ],
+)
+def test_bbox_eq_true(bbox1: BBox, bbox2: BBox) -> None:
     assert bbox1 == bbox2
-    assert bbox1 != bbox3
-    assert bbox1 != bbox4
-    assert bbox1 is not None
 
 
-def test_transform() -> None:
-    bbox1 = BBox([46.07, 13.23, 46.24, 13.57], CRS.WGS84)
-    bbox2 = bbox1.transform(CRS.POP_WEB).transform(CRS.WGS84)
+@pytest.mark.parametrize(
+    "bbox1, bbox2",
+    [
+        pytest.param(BBox((0, 0, 1, 1), CRS(1234)), (0, 0, 1, 1), id="different_types"),
+        pytest.param(BBox((0, 0, 1, 1), CRS(1234)), BBox((0, 0, 1, 1), CRS(4321)), id="different_CRS"),
+        pytest.param(BBox((0, 0, 1, 1), CRS(1234)), BBox((0, 0.00000001, 1, 1), CRS(1234)), id="different_coords"),
+    ],
+)
+def test_bbox_eq_false(bbox1: BBox, bbox2: BBox) -> None:
+    assert bbox1 != bbox2
+
+
+def test_bbox_transform() -> None:
+    original_bbox = BBox((46.07, 13.23, 46.24, 13.57), CRS.WGS84)
+    transformed_bbox = original_bbox.transform(CRS.POP_WEB)
+
+    assert transformed_bbox.crs == CRS.POP_WEB
+    assert list(transformed_bbox) == approx([5128488.941, 1486021.486, 5147413.254, 1524929.4087], rel=1e-10)
+
+    reconstructed_bbox = transformed_bbox.transform(CRS.WGS84)
 
-    for coord1, coord2 in zip(bbox1, bbox2):
-        assert coord1 == approx(coord2, abs=1e-8)
-    assert bbox1.crs == bbox2.crs
+    assert list(original_bbox) == approx(list(reconstructed_bbox), rel=1e-10)
+    assert original_bbox.crs == reconstructed_bbox.crs
 
 
-def test_transform_bounds() -> None:
-    bbox1 = BBox([46.07, 13.23, 46.24, 13.57], CRS.WGS84)
-    utm_crs = get_utm_crs(*bbox1.middle, source_crs=CRS.WGS84)
-    bbox2 = bbox1.transform_bounds(utm_crs).transform_bounds(CRS.WGS84)
+def test_bbox_transform_bounds() -> None:
+    original_bbox = BBox((46.07, 13.23, 46.24, 13.57), CRS.WGS84)
+    utm_crs = get_utm_crs(*original_bbox.middle, source_crs=CRS.WGS84)
+    reconstructed_bbox = original_bbox.transform_bounds(utm_crs).transform_bounds(CRS.WGS84)
 
-    assert bbox2.geometry.contains(bbox1.geometry)
-    assert bbox2.geometry.difference(bbox1.geometry).area > 1e-4
+    assert reconstructed_bbox.geometry.contains(original_bbox.geometry)
 
+    area_diff = reconstructed_bbox.geometry.difference(original_bbox.geometry).area
+    expected_diff = reconstructed_bbox.geometry.area / 20  # the area difference for this case is about 2.5%
+    assert area_diff < expected_diff
 
-def test_geometry() -> None:
-    bbox = BBox([46.07, 13.23, 46.24, 13.57], CRS.WGS84)
-    assert isinstance(bbox.get_geojson(), dict)
+
+def test_bbox_geometry_attribute() -> None:
+    bbox = BBox((0, 0, 1, 1), CRS.WGS84)
     assert isinstance(bbox.geometry, shapely.geometry.Polygon)
+    assert bbox.geometry.equals(shapely.geometry.Polygon([[0, 0], [0, 1], [1, 1], [1, 0], [0, 0]]))
 
 
-def test_buffer() -> None:
-    bbox = BBox([46.07, 13.23, 46.24, 13.57], CRS.WGS84)
+def test_bbox_buffer() -> None:
```

Comment:
Could this test case be parameterized like (test bbox, relative, expected ) 
I would also like to add simple bbox (0, 0, 10, 10) or sth.

Reply:
the test cases are quite different for 'relative' and 'not relative'.

The BBox is intentionally a bit ugly so that it doesn't only work on pretty cases :) But the calculations are simple enough here that one can doublecheck they are correct.

Reply:
we could parametrize the test BBox to have one pretty and one ugly :thinking: